### PR TITLE
EVG-5229 Do not run teardown on task-spawned hosts

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: units/host_termination.go [hunk 1]
```diff
@@ -227,36 +227,33 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
...
+				"operation":  "host termination issue",
+				"sender_err": serr,
+				"error":      err,
+				"host":       j.host.Id,
+				"subject":    subj,
+			})
+		} else {
+			mailer.Send(message.NewEmailMessage(level.Error, message.Email{
+				From:       settings.Notify.SMTP.From,
+				Recipients: settings.Notify.SMTP.AdminEmail,
```
Comment (richardsamuels) [inline units/host_termination.go]: I'd recommend checking if settings.Notify.SMTP.AdminEmail is empty

File: units/host_termination.go [hunk 2]
```diff
@@ -227,36 +227,33 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
...
+		subj := fmt.Sprintf("%v Error running teardown for host %v",
+			teardownFailurePreface, j.host.Id)
+
+		mailer, serr := j.env.GetSender(evergreen.SenderEmail)
+		if serr != nil {
+			grip.Alert(message.Fields{
+				"message":    "problem getting sender",
+				"operation":  "host termination issue",
+				"sender_err": serr,
+				"error":      err,
```
[the remark this answers is not in the document]
Comment (bsamek) [reply units/host_termination.go]: True, though this is a different condition, and at a higher logging level, which is currently above our slack threshold. I'd like to keep this.

File: units/host_termination.go (hunk 1 above)
Comment (bsamek) [reply units/host_termination.go]: Thanks, doing this.

File: units/host_termination.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (richardsamuels) [reply units/host_termination.go]: Fair enough, but should you be doing {s,}err.Error()?

**Pull request merged**